main/src/main/scala/sbt/internal/server/NetworkChannel.scala

```diff
@@ -53,6 +56,7 @@ final class NetworkChannel(
   private val VsCode = sbt.protocol.Serialization.VsCode
   private val VsCodeOld = "application/vscode-jsonrpc; charset=utf8"
   private lazy val jsonFormat = new sjsonnew.BasicJsonProtocol with JValueFormats {}
+  private val onGoingRequests: mutable.Set[String] = mutable.Set()
```

Comment:
`on...` is confusing with event handler like `onClick(...)`. Let's call this `pendingRequests`. Also it might be useful to keep the entire Request objects since it shouldn't add up in memory.

Reply:
done